substrate/pkg/apis/v1alpha1/substrate_validation.go

```diff
@@ -0,0 +1,28 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+func (s *Substrate) Validate(ctx context.Context) (errs *apis.FieldError) {
+	if len(s.Name) == 0 {
+		return errs.Also(apis.ErrMissingField("name"))
+	}
```

Comment:
I think even the name can be optional, we can default to something like `substrate` or `kit-substrate`

Reply:
Agreed. Let's get to this later. This isn't even wired up at the mo.